src/Hooks/Database/Support/QueryManager.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the tenancy/tenancy package.
+ *
+ * Copyright Tenancy for Laravel
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://tenancy.dev
+ * @see https://github.com/tenancy
+ */
+
+namespace Tenancy\Hooks\Database\Support;
+
+use Closure;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Database\QueryException;
+
+class QueryManager
```

Comment:
Let's make sure this implements and interface and register this class in the container

Reply:
That seems a bit overkill, but I suppose we could do that.  I'm already pulling it out of the container with `app(...)`, so it's easy enough to replace what class is used without needing the interface and explicit registration.

Reply:
It's not really overkill imo, if I need something that provides different logic / more features, I can 😄

Reply:
Right, that was my point though.  You already can